### PR TITLE
Jetack plugin: refactor connect screen to prevent duplicate

### PR DESCRIPTION
**Pull request by monsieur-z**
## Proposed changes:

The base connection screens in the Jetpack plugin and My Jetpack package are almost similar. This PR makes the body of the connection screen in the My Jetpack package available and uses that component in the Jetpack plugin.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

See this comment: https://github.com/Automattic/jetpack/issues/35485#issuecomment-1946836406

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

- Spin up a test site with this branch
- Connect the site from these two places and check that everything works as before:
  - `/wp-admin/admin.php?page=jetpack#/connect-user`
  - `/wp-admin/admin.php?page=my-jetpack#/connection`

### CODE REVIEW

File: projects/packages/my-jetpack/_inc/components/connection-screen/body.tsx [hunk 1]
```diff
@@ -3,33 +3,23 @@ import { ConnectScreen } from '@automattic/jetpack-connection';
...
 
 	return (
 		<ConnectScreen
-			title={ __(
-				'Unlock all the amazing features of Jetpack by connecting now',
-				'jetpack-my-jetpack'
-			) }
 			buttonLabel={ __( 'Connect your user account', 'jetpack-my-jetpack' ) }
 			loadingLabel={ __( 'Connecting your account…', 'jetpack-my-jetpack' ) }
-			apiRoot={ apiRoot }
```
Comment (monsieur-z) [inline projects/packages/my-jetpack/_inc/components/connection-screen/body.tsx:26]: Move these props to the parent component (`ConnectionScreen`) so the body can be reused.

**Comment by sergeymitr** [comment]
Hi @monsieur-z 👋 
Since #36508 is ready to be merged, should we switch this PR's destination branch from `refactor/connect-screen-ts` to `trunk`?

**Review by sergeymitr — DISMISSED** [review]
Works as expected, and the code looks a lot cleaner 👍

**Review by sergeymitr — APPROVED** [review]
Re-approving.

**Pull request merged**